Library/ViewModels/SearchViewModelTests.swift

```diff
@@ -53,6 +54,106 @@ internal final class SearchViewModelTests: TestCase {
+
+      self.vm.inputs.tapped(project: projects[0])
+
+      self.goToRefTag.assertValues([refTag])
+    }
+  }
+
+  func testSearchPopular_RefTag() {
+    let refTag = RefTag.searchPopular
+
+    let projects = (0...10).map { idx in .template |> Project.lens.id .~ (idx + 42) }
+    let response = .template |> DiscoveryEnvelope.lens.projects .~ projects
+
+    withEnvironment(apiService: MockService(fetchDiscoveryResponse: response)) {
+      self.vm.inputs.viewWillAppear(animated: false)
+
+      self.scheduler.advance()
+
+      self.hasProjects.assertValues([true], "Projects emitted immediately upon view appearing.")
+
+      self.vm.inputs.tapped(project: projects[8])
+
+      self.goToRefTag.assertValues([refTag])
+    }
+  }
+
+  func testSearchFeatured_RefTag() {
+    let refTag = RefTag.searchFeatured
+
+    let projects = (0...10).map { idx in .template |> Project.lens.id .~ (idx + 42) }
+    let response = .template |> DiscoveryEnvelope.lens.projects .~ projects
+    let projects2 = (20...30).map { idx in .template |> Project.lens.id .~ (idx + 42) }
+    let response2 = .template |> DiscoveryEnvelope.lens.projects .~ projects2
+
+    withEnvironment(apiService: MockService(fetchDiscoveryResponse: response)) {
+      self.vm.inputs.viewWillAppear(animated: false)
+
+      self.scheduler.advance()
+
+      self.hasProjects.assertValues([true], "Projects emitted immediately upon view appearing.")
+
+      withEnvironment(apiService: MockService(fetchDiscoveryResponse: response2)) {
+
+        self.vm.inputs.searchFieldDidBeginEditing()
+        self.vm.inputs.searchTextChanged("robots")
+
+        self.scheduler.advance()
+        self.hasProjects.assertValues([true, false, true],
+                                      "Projects emitted immediately upon view appearing.")
+
+        self.vm.inputs.tapped(project: projects2[0])
+
+        self.goToRefTag.assertValues([refTag])
+      }
+    }
+  }
+
+  func testSearch_RefTag() {
+    let refTag = RefTag.search
+
+    let projects = (0...10).map { idx in .template |> Project.lens.id .~ (idx + 42) }
+    let response = .template |> DiscoveryEnvelope.lens.projects .~ projects
+    let projects2 = (20...30).map { idx in .template |> Project.lens.id .~ (idx + 42) }
+    let response2 = .template |> DiscoveryEnvelope.lens.projects .~ projects2
+
+    withEnvironment(apiService: MockService(fetchDiscoveryResponse: response)) {
+      self.vm.inputs.viewWillAppear(animated: false)
+
+      self.scheduler.advance()
+
+      self.hasProjects.assertValues([true], "Projects emitted immediately upon view appearing.")
+
+      withEnvironment(apiService: MockService(fetchDiscoveryResponse: response2)) {
+
+        self.vm.inputs.searchFieldDidBeginEditing()
+        self.vm.inputs.searchTextChanged("robots")
+
+        self.scheduler.advance()
+        self.hasProjects.assertValues([true, false, true],
+                                      "Projects emitted immediately upon view appearing.")
```

Comment:
I think we could prob remove the `hasProjects` assertions since we have asserted on it in other tests and this test is mostly focused on the ref tag.

we could make the whole user script a bit succinct with:

```swift
self.vm.inputs.searchFieldDidBeginEditing()
self.vm.inputs.searchTextChanged("robots")
self.scheduler.advance()
self.vm.inputs.tapped(project: projects2[2])

self.goToRefTag.assertValues([refTag])
```

can do something similar in the other tests too